portfolio/src/main/webapp/script.js

```diff
@@ -45,3 +45,96 @@ function swapHidden(id1,id2) {
     toBeRevealed.style.display = 'inline-block';
     toBeHidden.style.display = 'none';
 }
+
+/*Below is code that handles requests to the data servlet.*/
+
+// Create the intro page using the JSON.
+async function load() {
+    const response = await fetch('/data');
+    const data = await response.json();
+    loadIntro(data[0]);
+    loadComments(data[1][0]);
+}
+
+async function loadComments(commentData) {
+    const commentSection = document.getElementById('comment-box');
+    for(i=0; i<commentData.length; i++) {
+        const comment = commentData[i].comment;
+        const author = commentData[i].username;
+        const wrapper = document.createElement('div');
+        wrapper.classList.add("comment-style");
+        wrapper.appendChild(
+            createH2(author)
+        );
+        wrapper.appendChild(
+            createP("Comment: " + comment)
+        );
+        commentSection.appendChild(wrapper);
+    }
+}
+
+// Creates the intro page to the portfolio.
+function loadIntro(pageData) {
+    const intro = document.getElementById('welcome');
+
+    //Append the intro image.
+    intro.appendChild(
+        createImg(pageData[0])
+    );
+
+    //Append "Welcome to my Portfolio!"
+    intro.appendChild(
+        createH2(pageData[3])
+    );
+
+    //Append the p element that is on the intro page.
+    intro.appendChild(
+        createP(pageData[1])
+    );
+    
+    //Append the button that navigates to the following page
+    btn = createBtn(pageData[2]);
+    intro.appendChild(btn);
+    btn.onclick = function() {
+        (swapHidden('welcome','intro-main'));
+    };
+}
+
+//Creates an h2 element with the given parameters
+function createH2(text) {
+    const h2Element = document.createElement('h2');
+    h2Element.innerText = text;
+    return h2Element;
+}
+
+//Creates a p element with the given parameters
+function createP(text) {
+    const pElement = document.createElement('p');
+    pElement.innerText = text;
+    return pElement;
+}
+
+//Creates a button element with the given parameters
+function createBtn(text) {
+    const btn = document.createElement('button');
+    btn.innerText = text;
+    return btn;
+}
+
+//Creates an image with the given parameters
+function createImg(src) {
+    const img = document.createElement('img');
+    img.src = src;
+    return img;
+}
+/**
+ * Prints the welcome statement from the servlet.
+ * This code will be commented out in favor of a function that will parse the JSON received from the servlet.
+ * Like the case in the DataServlet.java file, this code is in a current PR and I am unsure as to whether it
+ * is okay to fully remove this code until its PR is merged.
+ */
+// function getWelcome() {
+//     fetch('/data').then(response => response.text()).then((welcome) => {
+//     document.getElementById('welcome-container').innerText = welcome;
+//   });
+// }
```

Comment:
What you could do is merge the PR that uses it first, and then remove it in this PR and you should be able to merge both cleanly without conflicts